lib/js/jsffi.nim

```diff
@@ -0,0 +1,479 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2017 Nim Authors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+## This Module implements types and macros to facilitate the wrapping of, and
+## interaction with JavaScript libraries. Using the provided types ``JsObject``
+## and ``JsAssoc`` together with the provided macros allows for smoother
+## interfacing with JavaScript, allowing for example quick and easy imports of
+## JavaScript variables:
+##
+## .. code-block:: nim
+##
+##  # Here, we are using jQuery for just a few calls and do not want to wrap the
+##  # whole library:
+##
+##  # import the document object and the console
+##  magicVar(document, JsObject)
+##  magicVar(console, JsObject)
+##  # import the "$" function
+##  proc jq(selector: JsObject): JsObject {. importcpp: "$(#)" .}
+##
+##  # Use jQuery to make the following code run, after the document is ready.
+##  # This uses an experimental ``.()`` operator for ``JsObject``, to emit
+##  # JavaScript calls, when no corresponding proc exists for ``JsObject``.
+##  proc main =
+##    jq(document).ready(proc() =
+##      console.log("Hello JavaScript!")
+##    )
+##
+## JsFFI also provides macros to facilitate fully statically typed interaction
+## with Javascript. To this end, one often has to write types, where every field
+## is marked with ``exportc`` to avoid mangling. For types like React attributes
```

Comment:
Would the fix needed be to forego mangling of fields and field accesses as done in `genFieldAddr` and `genFieldAccess` in module `jsgen` (changing the default mangling behaviour for only the js target)? If so, I'll try to fix this in a separate pull request and remove `pragmaTypeSection` from this one, if not, could you please point me in the right direction?